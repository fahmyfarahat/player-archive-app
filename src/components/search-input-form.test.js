import React from "react";
import { shallow } from "enzyme";
import SearchInputForm from "./search-input-form";
import toJson from "enzyme-to-json";

describe("SearchInputForm - general", () => {
 let wrapper;
  beforeEach(()=> {
    const props = {
      buttonText: "go",
      name: "input",
      placeholder: "text..",
      onSubmit: jest.fn()
    }
    wrapper = shallow(<SearchInputForm {...props}/>);
  });
  
  it("should render without crashing", () => {
    expect(wrapper).toHaveLength(1);
  });
    
  it("should match Snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
