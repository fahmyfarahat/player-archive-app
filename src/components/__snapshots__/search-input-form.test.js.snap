// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInputForm - general should match Snapshot 1`] = `
<SectionWrapper
  className="searchInput-section"
>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <InputGroup
      className="mb-3"
    >
      <FormControl
        name="input"
        onChange={[Function]}
        placeholder="text.."
        size="lg"
        type="text"
        value=""
      />
      <InputGroupAppend>
        <Button
          active={false}
          disabled={false}
          size="lg"
          type="submit"
          variant="primary"
        >
          go
        </Button>
      </InputGroupAppend>
    </InputGroup>
  </Form>
</SectionWrapper>
`;
